Source/Plugins/EventBroadcaster/EventBroadcaster.cpp

```diff
@@ -11,17 +11,85 @@
 #include "EventBroadcaster.h"
 #include "EventBroadcasterEditor.h"
 
-std::shared_ptr<void> EventBroadcaster::getZMQContext() {
-    // Note: C++11 guarantees that initialization of static local variables occurs exactly once, even
-    // if multiple threads attempt to initialize the same static local variable concurrently.
+EventBroadcaster::ZMQContext* EventBroadcaster::sharedContext = nullptr;
+CriticalSection EventBroadcaster::sharedContextLock{};
+
+EventBroadcaster::ZMQContext::ZMQContext(const ScopedLock& lock)
 #ifdef ZEROMQ
-    static const std::shared_ptr<void> ctx(zmq_ctx_new(), zmq_ctx_destroy);
-#else
-    static const std::shared_ptr<void> ctx;
+    : context(zmq_ctx_new())
+#endif
+{
+    sharedContext = this;
+}
+
+// ZMQContext is a ReferenceCountedObject with a pointer in each instance's 
+// socket pointer, so this only happens when the last instance is destroyed.
+EventBroadcaster::ZMQContext::~ZMQContext()
+{
+    ScopedLock lock(sharedContextLock);
+    sharedContext = nullptr;
+#ifdef ZEROMQ
+    zmq_ctx_destroy(context);
+#endif
+}
+
+void* EventBroadcaster::ZMQContext::createZMQSocket()
+{
+#ifdef ZEROMQ
+    jassert(context != nullptr);
+    return zmq_socket(context, ZMQ_PUB);
+#endif
```

Comment:
Realized soon after the last commit that there should be a return statement for the case that ZEROMQ isn't defined (would just return `nullptr`). I have a feeling this would produce at least a warning on Linux.